### PR TITLE
MDS-2094 Code Required Reports form updates

### PR DESCRIPTION
**Pull request by JustinMacaulay**
A new table component to be used when displaying documents on forms.

### CODE REVIEW

File: frontend/src/components/Forms/reports/ReportSubmissions.js [hunk 1]
```diff
@@ -17,6 +17,23 @@ const defaultProps = {
   mineReportSubmissions: [],
 };
 
+const updateSubmissionHandler = (mine_document_guid, props) => {
+  const fileToRemove = props.mineReportSubmissions[
+    props.mineReportSubmissions.length - 1
+  ].documents.filter((doc) => doc.mine_document_guid === mine_document_guid)[0];
+  let updatedSubmissions = props.mineReportSubmissions;
+  console.log(fileToRemove);
+  updatedSubmissions[updatedSubmissions.length - 1].documents = reject(
```
[the remark this answers is not in the document]
Comment (JustinMacaulay) [reply frontend/src/components/Forms/reports/ReportSubmissions.js]: Removed

File: frontend/src/components/common/UploadedDocumentTable.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import { Table } from "antd";
+import PropTypes from "prop-types";
+import CustomPropTypes from "@/customPropTypes";
+import LinkButton from "@/components/common/LinkButton";
+import { downloadFileFromDocumentManager } from "@/utils/actionlessNetworkCalls";
+
+/**
+ * @class  MinePermitInfo - contains all permit information
+ */
+
```
[the remark this answers is not in the document]
Comment (JustinMacaulay) [reply frontend/src/components/common/UploadedDocumentTable.js]: removed

File: frontend/src/components/Forms/reports/ReportSubmissions.js [hunk 3]
```diff
@@ -17,6 +17,22 @@ const defaultProps = {
   mineReportSubmissions: [],
 };
 
+const updateSubmissionHandler = (mine_document_guid, props) => {
+  const fileToRemove = props.mineReportSubmissions[
+    props.mineReportSubmissions.length - 1
+  ].documents.filter((doc) => doc.mine_document_guid === mine_document_guid)[0];
+  let updatedSubmissions = props.mineReportSubmissions;
```
Comment (Jsyro) [inline frontend/src/components/Forms/reports/ReportSubmissions.js]: Should this be a find() as there should only be on item in the list that this returns right? filter implies many.

Comment (JustinMacaulay) [reply frontend/src/components/Forms/reports/ReportSubmissions.js]: Changed and yeah this is a way better option.

**Comment by JustinMacaulay** [comment]
Closing this PR asall the changes are now wrapped up in ryans PR 1032

**Pull request closed without merging**